src/Entity/Address.php

```diff
@@ -49,9 +49,20 @@ class Address
      */
     private $city;
 
-    public function __construct()
+    public function __construct(
+        string $name,
+        string $streetAddress,
+        string $streetAddressComplementary,
+        string $postalCode,
+        string $city
+    )
     {
         $this->id = Uuid::uuid4()->toString();
```

Comment:
I think we should create the Uuid in the caller not in this class directly, and passed as parameter as well.

Reply:
I will do this if you like, however this will be unnecessary code duplication

Reply:
How so? I would guess the object should. always somehow be created from a factory or a builder, probably used in the repository or something like that, and that's where I would generate the id.

Reply:
Sorry, here's some literature to support my weird way to express my thoughts 😄 
https://matthiasnoback.nl/2018/05/when-and-where-to-determine-the-id-of-an-entity/